### PR TITLE
test: Include a cluster test for Request Validator

### PR DESCRIPTION
**Pull request by claudiachua**
Internal JIRA link: https://issues.corp.twilio.com/browse/DI-2020

### CODE REVIEW

File: cluster_test.go [hunk 1]
```diff
@@ -119,3 +126,133 @@ func TestListParams(t *testing.T) {
...
+}
+
+func executeFlow(t *testing.T, flowSid string) {
+	_, exeErr := testClient.StudioV2.CreateExecution(flowSid, createStudioExecutionParams())
+	if exeErr != nil {
+		t.Fatal("Error with Studio Execution Creation: ", exeErr)
+	}
+}
+
+func TestRequestValidation_GETMethod(t *testing.T) {
```
Comment (childish-sambino) [inline cluster_test.go:204]: Think all the implementation here can be extracted into a separate function where you just pass the `method` that you want to test.

File: cluster_test.go [hunk 2]
```diff
@@ -119,3 +126,133 @@ func TestListParams(t *testing.T) {
...
+	resp, flowErr := testClient.StudioV2.CreateFlow(params)
+	if flowErr != nil {
+		t.Fatal("Error with Studio Flow Creation: ", flowErr)
+	}
+	flowSid := *resp.Sid
+	executeFlow(t, flowSid)
+
+	//Await for Request Validation
+	assert.True(t, <-channel)
+	defer testClient.StudioV2.DeleteFlow(flowSid)
```
Comment (childish-sambino) [inline cluster_test.go:229]: If the assertion fails or the flow fails to execute, does the flow still get deleted?

Comment (claudiachua) [reply cluster_test.go:229]: If request validator returns false, the flow still get deleted. Will add in the logic to teardown if the execution of http Request in flow fails

File: cluster_test.go [hunk 3]
```diff
@@ -120,6 +127,127 @@ func TestListParams(t *testing.T) {
...
+	executeFlow(t, flowSid)
+
+	//Await for Request Validation
+	afterCh := time.After(5 * time.Second)
+	select {
+	case validate := <-channel:
+		assert.True(t, validate)
+	case <-afterCh:
+		t.Fatal("No request was sent to validation server")
+	}
```
Comment (claudiachua) [inline cluster_test.go:238]: Relying on a timeout of 5 seconds if no request hit the validation server. This is a workaround as StudioV2.FetchExecutionContext was not working as expected.

**Review by JenniferMah — APPROVED** [review]

**Review by beebzz — APPROVED** [review]

**Pull request merged**